### DISCUSSION TITLE
Use Math Expression value in PromQL Query

### DISCUSSION THREAD

**Discussion by manfred-ackermann** [grafana/grafana#53676] [Alerting]
I like to setup my Grafana alerts like

1. Declare a "threshold" in a Math Expression
2. Then the alert query: avg_over_time(...)
3. The Reduce Expression
4. The trigger Math Expression that is evaluated

It would be awesome if I could use the value of the threshold expression in the alert query.

So instead of
```
avg_over_time(
  {source="Akamai"} | json |  __error__ = ""
  | turnAroundTimeMSec > 2500
  | unwrap turnAroundTimeMSec [1m]
) by (reqHost,pathClass)
```
I'd like to 
`| turnAroundTimeMSec > ${threshold}`
or
`| turnAroundTimeMSec > {{$values.threshold.Value}}`
to have one and one place only where I can consistent declare my values.

Works perfect for the alert description
`TurnAroundTime on host {{$labels.reqHost}}/{{$labels.pathClass}} exceeds {{$values.threshold.Value}}ms with {{$values.ms.Value}}ms. `, only on the query I have to maintain this value redundant.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by manfred-ackermann** [comment]
https://github.com/grafana/grafana/issues/82888
